[WIP] Upgrade ProductDownloader component

ztanner opened the pull request:
Updates the downloads page to use the new ProductDownloader component.

_this is a WIP_ pending copy & UI review

/cc @BrandonRomano

jescalan commented:
I think we need these to be filled in:

![](https://p176.p0.n0.cdn.getcloudapp.com/items/8LunoEnp/Screen%20Shot%202020-12-14%20at%2012.15.25%20PM.png?source=viewer&v=9adbea1652d100ea558397467af5e1b2)

jescalan commented on website/package.json:
```diff
@@ -4,6 +4,7 @@
   "version": "1.0.0",
   "author": "HashiCorp",
   "dependencies": {
+    "@hashicorp/mktg-assets": "^2.0.0-alpha.0",
```
What's going on with this dependency add?

ztanner replied:
Using it for the logos here - it seemed like the easiest, most consistent way to get the colored logo into all of these PRs (this PR is duplicated across the various other products)

jescalan commented on website/package.json:
```diff
@@ -12,10 +13,11 @@
     "@hashicorp/react-search": "^2.1.0",
     "@hashicorp/react-section-header": "2.0.0",
     "@hashicorp/react-subnav": "3.2.3",
+    "@hashicorp/react-tabs": "^2.0.0",
```
This too - if its a dependency of `product-downloader`, it should be automatically brought in by that component

ztanner replied:
Was having trouble here -- it is a dependency of product-downloader, but is unstyled unless the host app (Packer, in this case) imports the styles in style.css. Curious if you had any thoughts about how to fix this.

ztanner commented:
@jescalan  Definitely! I believe @BrandonRomano is working on getting the copy here -- these are just WIP PRs. It might be good to add a label, but I'm not able to since I'm not a collaborator.

jescalan commented on website/pages/downloads/index.jsx:
```diff
@@ -1,25 +1,102 @@
+              'sudo yum -y install packer',
+            ],
+            os: 'linux',
+          },
```
Can we move this out to the `version` data file like it is here?

https://github.com/hashicorp/waypoint/blob/main/website/data/version.js#L6

ztanner replied:
> Ah ok - I adjusted the title to reflect this. Sorry, when I was tagged I thought it was ready for review, but I see now that you were not the one who tagged me haha

Good call - I've updated the title on the other PRs as well.

jescalan reviewed (changes requested):
Looking good, just a couple things I think we should address before merge:

- [ ] Move extra dependencies into product downloader itself
- [ ] Move package manager data into data file
- [ ] Double check with packer team and engineering services that the package managers shown here are officially maintained
- [ ] Make sure that the placeholder content is replaced by actual content

jescalan commented:
Ah ok - I adjusted the title to reflect this. Sorry, when I was tagged I thought it was ready for review, but I see now that you were not the one who tagged me haha

ztanner commented:
> Ah ok - I adjusted the title to reflect this. Sorry, when I was tagged I thought it was ready for review, but I see now that you were not the one who tagged me haha

Good call - I've updated the title on the other PRs as well.

azr commented:
Hello there, what's the status of this one; should we merge it ?

jescalan commented:
Good question - I think maybe @BrandonRomano could answer this best right now

nywilken commented:
@jescalan after your recent change to the download page is it safe to close this PR?

jescalan commented:
Oh yes, so sorry this lingered for so long!
